pages/signup/index.js

```diff
@@ -1,126 +1,262 @@
 import React, { useState, useEffect } from 'react'
 import Image from 'next/image'
 import { useDispatch, useSelector } from 'react-redux'
-import { useRouter } from 'next/router'
-import { changeLoginType, signup } from 'redux/reducers/authSlice'
+import Router from 'next/router'
+import {} from 'redux/reducers/authSlice'
+import { getSession } from 'next-auth/react'
+
+import { REGISTER_USER } from 'utils/mutations'
+import { useMutation } from '@apollo/client'
+import { MOON_HOLDINGS } from 'app/constants/copy'
+import BannerModal from 'components/modals/BannerModal'
 
 const Index = () => {
-  const router = useRouter()
   const dispatch = useDispatch()
   const [email, setEmail] = useState('')
   const [password, setPassword] = useState('')
   const [confirmPassword, setConfirmPassword] = useState('')
+  const [message, setMessage] = useState('')
+  const [error, setError] = useState(false)
+  const [showModal, setShowModal] = useState(false)
+
+  const [signUp, { loading, data }] = useMutation(REGISTER_USER)
+
+  useEffect(() => {
+    if (data) {
+      setModal(
+        'You have successfully signed up. Please verify your email to login.',
+        false,
+        true
+      )
+      const timer = setTimeout(() => {
```

Comment:
Why need setTimeout here?

Reply:
to wait a few secs before routing to login instead of routing right away